### PR TITLE
Remove usage of globals from 08-wasm

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Removes the globals in ibcwasm and restructures module to move more functionality over to the keeper. This was previously present in `types/` due to limitations imposed by having all calls flow through the client state.

closes: https://github.com/cosmos/ibc-go/issues/5833
closes: https://github.com/cosmos/ibc-go/issues/5834

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/keeper/keeper.go [hunk 1]
```diff
@@ -2,33 +2,40 @@ package keeper
...
+	checksums    collections.KeySet[[]byte]
 	storeService store.KVStoreService
-
+	// handling queries
+	queryPlugins QueryPlugins
+	cdc          codec.BinaryCodec
 	clientKeeper types.ClientKeeper
-
-	authority string
+	authority    string
```
Comment (crodriguezvega) [inline modules/light-clients/08-wasm/keeper/keeper.go:38]: Just for readability: maybe add an empty line to separate these? Or move them to the top?

File: modules/light-clients/08-wasm/keeper/keeper.go [hunk 2]
```diff
@@ -41,7 +49,41 @@ func (k Keeper) GetAuthority() string {
...
 }
 
-func (Keeper) storeWasmCode(ctx sdk.Context, code []byte, storeFn func(code wasmvm.WasmCode, gasLimit uint64) (wasmvm.Checksum, uint64, error)) ([]byte, error) {
+// Logger returns a module-specific logger.
+func (Keeper) Logger(ctx sdk.Context) log.Logger {
+	return moduleLogger(ctx)
+}
+
+func moduleLogger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", "x/"+exported.ModuleName+"-"+types.ModuleName)
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/keeper/keeper.go:57]: I think it's fine like this.

**Review by crodriguezvega — APPROVED** [review]
Amazing work, @DimitrisJim. This is really an exhaustive and nice refactoring.

I agree that we should move a bunch of types, variables and constants in `types` that are only used in `keeper` to an internal package. Happy to write an issue for this a tackle it a bit later (but before v9 release).

We should also add an issue to update documentation and migration docs (I guess for v0.3.x of 08-wasm with ibc-go v9.0) with these changes and any subsequent changes in future PRs. We can also tackle this later on.

**Review by damiannolan — APPROVED** [review]
Overall LGTM! Excellent work on the refactor :) 

I left some comments and questions we should address before merge

**Review by chatton — APPROVED** [review]
Awesome work with this, thanks for taking on this monster LGTM

Happy to have some of the other comments addressed in a follow up :vulcan_salute:

File: modules/light-clients/08-wasm/types/gas_register.go [hunk 3]
```diff
@@ -63,10 +63,13 @@ const (
...
-var defaultPerByteUncompressCost = wasmvmtypes.UFraction{
-	Numerator:   15,
-	Denominator: 100,
-}
+var (
+	defaultPerByteUncompressCost = wasmvmtypes.UFraction{
+		Numerator:   15,
+		Denominator: 100,
+	}
+	VMGasRegister = NewDefaultWasmGasRegister()
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/types/gas_register.go:71]: Yes, I did sync this when upgraded to wasmvm v2.

**Pull request merged**